src/collocation.jl

```diff
@@ -10,7 +10,16 @@ end
 function BVPSystem(fun, bc, p, x, y, order)
     T, U = eltype(x), eltype(y)
     M, N = size(y)
-    BVPSystem{T,U}(order, M, N, fun, bc, p, x, y, vector_alloc(T, M, N), vector_alloc(T, M, N), eltype(y)(M))
+    BVPSystem{T,U}(order, M, N, fun, bc, p, x, y, vector_alloc(U, M, N), vector_alloc(U, M, N), typeof(x)(M))
+end
+
+# Dispatch aware of eltype(x) != eltype(prob.u0)
+function BVPSystem(prob::BVProblem, x, order)
+    U = eltype(prob.u0)
+    M = length(prob.u0)
+    N = size(x,1)
+    y = vector_alloc(U, M, N)
+    BVPSystem(order, M, N, prob.f, prob.bc, prob.p, x, y, vector_alloc(U, M, N), vector_alloc(U, M, N), typeof(x)(undef,M))
```

Comment:
Please change the variable name from `U` to `T` to avoid confusion.

Reply:
I think that `T` is still confusing here, since it has always corresponded to `eltype(x)`, don't you think that one should choose something free, e.g. `Y` so that `U` is `Vector{Y}`, just to indicate that `Y ≠ T` is possible?

Reply:
It's pretty standard to use `T` as the `eltype` of an array.